File: backend/src/main/java/me/iamhardyha/bugbuddy/service/UserService.java
package me.iamhardyha.bugbuddy.service;

import me.iamhardyha.bugbuddy.global.exception.BugBuddyException;
import me.iamhardyha.bugbuddy.global.response.ErrorCode;
import me.iamhardyha.bugbuddy.model.entity.UserEntity;
import me.iamhardyha.bugbuddy.repository.UserRepository;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
@Transactional(readOnly = true)
public class UserService {

    private final UserRepository userRepository;

    public UserService(UserRepository userRepository) {
        this.userRepository = userRepository;
    }

    public UserEntity findById(Long id) {
        return userRepository.findById(id)
                .orElseThrow(() -> new BugBuddyException(ErrorCode.USER_NOT_FOUND));
    }
}

File: backend/src/main/kotlin/me/iamhardyha/bugbuddy/service/UserService.kt
package me.iamhardyha.bugbuddy.service

import me.iamhardyha.bugbuddy.global.exception.BugBuddyException
import me.iamhardyha.bugbuddy.global.response.ErrorCode
import me.iamhardyha.bugbuddy.model.entity.UserEntity
import me.iamhardyha.bugbuddy.repository.UserRepository
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
@Transactional(readOnly = true)
class UserService(
    private val userRepository: UserRepository,
) {
    fun findById(id: Long): UserEntity =
        userRepository.findById(id).orElseThrow { BugBuddyException(ErrorCode.USER_NOT_FOUND) }
}
